l10n_it_asset_management/models/asset_accounting_info.py

```diff
@@ -0,0 +1,220 @@
+# Author(s): Silvio Gregorini (silviogregorini@openforce.it)
+# Copyright 2019 Openforce Srls Unipersonale (www.openforce.it)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AssetAccountingInfo(models.Model):
+    """
+    This model is necessary to manage info about the relationships between
+    assets and accounting records. We could have used `Many2many` fields to
+    create tables between assets, depreciation lines, account moves and account
+    move lines; but we need a custom
+    management, a bit more complex than the one provided by the standard
+    `Many2many` field.
+
+    NB: a few methods could have been decorated with @api.constrains or
+    @api.depends, but that would have limited them by being triggered only
+    upon some fields' changes, and would have made it very hard to be
+    customizable by third party modules.
+    Therefore, it has been chosen to define them as simple Python methods which
+    are called right after ``create`` and ``write`` methods if some fields
+    (which can be retrieved by method ``get_main_fields``) are found within
+    ``vals`` dictionary.
+    """
+
+    _name = "asset.accounting.info"
+    _description = "Asset Accounting Relations"
+    _table = "asset_accounting_info"
+
+    asset_id = fields.Many2one("asset.asset", ondelete="set null", string="Asset")
+
+    company_id = fields.Many2one(
+        "res.company",
+        readonly=True,
+        string="Company",
+    )
+
+    dep_line_id = fields.Many2one(
+        "asset.depreciation.line", ondelete="set null", string="Depreciation Line"
+    )
+    move_type = fields.Selection(related="dep_line_id.move_type")
+    move_id = fields.Many2one("account.move", ondelete="set null", string="Move")
+
+    move_line_id = fields.Many2one(
+        "account.move.line", ondelete="set null", string="Move Line"
+    )
+
+    relation_type = fields.Selection(
+        [
+            ("create", "Asset Creation"),
+            ("update", "Asset Update"),
+            ("partial_dismiss", "Asset Partial Dismissal"),
+            ("dismiss", "Asset Dismissal"),
+        ],
+        required=True,
+    )
+
+    @api.model
+    def create(self, vals):
```

Comment:
Dovresti usare il decoratore `@api.model_create_multi` e modificare la funzione per warning `is not overriding the create method in batch`